### PR TITLE
Adrien/docker+upped transifex cli

### PR DESCRIPTION
**Pull request by why-not-try-calmer**
**updated**:
- Dockerized local deployment, increasing reproducibility
- Updated CLI to Go, API v3. compliant
- Added pre-commit + pre-commit hook
- Updated the README.

**original**: 
Dockerizing the docs. Review appreciated. The container features the new transifex cli binary, so the container could be re-used for the issue of deploying the docs with that transifex cli.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+RUN apt-get update && apt upgrade -y && pip install pipenv
+WORKDIR /opt/app
+COPY ./requirements.txt .
+ENV PIPENV_VENV_IN_PROJECT=1
+RUN pipenv install --three
+
+# Build the transifex-cli go static binary
+FROM golang:1-bullseye as go-builder
+WORKDIR /opt/app
+RUN git clone https://github.com/transifex/cli . && make build
```
Comment (suricactus) [inline Dockerfile:12]: what if their master is currently broken? Can we install either a fixed version or the latest stable version?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+# Gather the build artifacts
+FROM python:3.9.13-slim-bullseye
+WORKDIR /opt/app
+COPY . .
+COPY --from=python-builder /opt/app/.venv/ .venv/
+COPY --from=go-builder /opt/app/bin/tx /bin/
+CMD . .venv/bin/activate && mkdocs serve -a 0.0.0.0:8000
```
Comment (suricactus) [inline Dockerfile:20]: Small styling comment, but newlines at the end of the files are highly appreciated. :)

Comment (suricactus) [reply Dockerfile:20]: Actually, you can add this as a pre-commit hook. Browse our repos where you can see we use `pre-commit` heavily.
Example configuration about this:

`.pre-commit-config.yml`
```
repos:
  # Fix end of files
  - repo: https://github.com/pre-commit/pre-commit-hooks
    rev: v3.2.0
    hooks:
      - id: trailing-whitespace
      - id: end-of-file-fixer
      - id: mixed-line-ending
        args:
          - '--fix=lf'

```

Comment (suricactus) [reply Dockerfile:20]: Bonus points: check this via GitHub actions too.

Comment (suricactus) [reply Dockerfile:24]: Should be enough now. Personally I am not a huge fan of enforcing checks before running something, as you might want to test something quick and dirty, or have to apply immediate change without worrying too much about styling.

Adding a bit of docs how to install pre-commit for local development would be great too.

File: Dockerfile (hunk 1 above)
Comment (suricactus) [reply Dockerfile]: Thanks.

File: README.md [hunk 3]
```diff
@@ -92,14 +96,36 @@ If you want more information about forking you can find it
...
 pipenv run mkdocs serve
 ```
 
-The local doc will be available at http://localhost:8000
+The local doc will be available at http://localhost:8000.
+
+To test your work against your pre-commit hook:
+
+    pipenv install pre-commit
+    pre-commit run --all-files
```
Comment (suricactus) [inline README.md:111]:
```suggestion
#### Contribute
Before commiting, install [pre-commit](https://pre-commit.com/) to auto-format your contributions.

```
pip install pre-commit
pre-commit install
```

Comment (why-not-try-calmer) [reply README.md]: Thank you very much, committed your suggestion, amended to pip install --user to minimize footprint on system-wide Python.

**Comment by why-not-try-calmer** [comment]
I apologize for destroying your mailboxes with the failed GitHub actions on commit/PR. Learned how to not do this anymore with [skip action] commit message. I've manage to make the failure happen where it should, i.e. the actual contents for the documentation. I'd be happy to move this forward depending on your input.

**Comment by suricactus** [comment]
For me it makes sense that we apply the required changes to make pre-commit happy, but let's see what @marioba thinks about it.

**Comment by marioba** [comment]
Yes, you can reformat anything you need to make pre-commit happy.

**Review by suricactus — APPROVED** [review]
Thanks, LGTM!

**Pull request merged**